sdk/core/src/main/java/com/klaviyo/coresdk/networking/KlaviyoProperties.kt

```diff
@@ -0,0 +1,112 @@
+package com.klaviyo.coresdk.networking
+
+import com.klaviyo.coresdk.utils.KlaviyoPreferenceUtils
+import java.io.Serializable
+
+/**
+ * Abstract class that wraps around a map to control access to its contents.
+ * Provides helper functions to add properties to the map while controlling the keys available for entry
+ */
+abstract class KlaviyoProperties {
+    internal val propertyMap: MutableMap<String, Serializable> = mutableMapOf()
+
+    /**
+     * Adds a new key/value pair to the map.
+     * [KlaviyoPropertyKeys] adds some control to what keys our property maps recognise
+     */
+    fun addProperty(propertyKey: KlaviyoPropertyKeys, value: String) {
+        propertyMap[propertyKey.name] = value
+    }
+
+    /**
+     * Adds a custom property to the map.
+     * Custom properties can define any key name that isn't already reserved by Klaviyo
+     */
+    abstract fun addCustomProperty(key: String, value: Serializable)
+
+    /**
+     * Fetches and returns our property map after converting it into a immutable map
+     */
+    internal fun toMap(): Map<Any, Any> {
+        return propertyMap.toMap()
+    }
+}
+
+/**
+ * Controls the data that can be input into a map of customer properties recognised by Klaviyo
+ */
+class KlaviyoCustomerProperties: KlaviyoProperties() {
+
+    init {
+        setAnonymousId()
```

Comment:
Crap sorry maybe this is bad too...this could block the main thread. Probably it was better before since that method is always access on the main thread.

Reply:
Ah damn you are right I didn't even consider this stupidly enough.

Although, since its a shared preference access, and our shared preferences file remains very very small, there's actually no need to keep this off the UI thread. It will not actually block execution cause its so simple

Reply:
But if you build up a large shared preferences file then it can block. So I still think this is smell that should change